model/span.go

```diff
@@ -77,15 +106,15 @@ func (s *Span) GetSpanKind() (spanKind trace.SpanKind, found bool) {
 }
 
 // GetSamplerType returns the sampler type for span
-func (s *Span) GetSamplerType() string {
+func (s *Span) GetSamplerType() SamplerType {
 	// There's no corresponding opentelemetry tag label corresponding to sampler.type
-	if tag, ok := KeyValues(s.Tags).FindByKey(samplerType); ok {
+	if tag, ok := KeyValues(s.Tags).FindByKey(keySamplerType); ok {
 		if tag.VStr == "" {
-			return samplerTypeUnknown
+			return toSamplerType[SamplerTypeUnrecognized.String()]
 		}
-		return tag.VStr
+		return toSamplerType[tag.VStr]
```

Comment:
```suggestion
		if s, ok := toSamplerType[tag.VStr]; ok {
				return s
		}
```

Reply:
and `if tag.VStr == "" {` above is not necessary, because having `sampler.type=""` in the span is very unlikely, but the map lookup for `""` will take care of it anyway

Reply:
Thanks again Yuri for this hint. 😄